prover/src/lib.rs

```diff
@@ -524,45 +527,10 @@ pub trait Prover {
         composition_poly_trace: CompositionPolyTrace<E>,
         num_constraint_composition_columns: usize,
         domain: &StarkDomain<Self::BaseField>,
-    ) -> (ConstraintCommitment<E, Self::HashFn, Self::VC>, CompositionPoly<E>)
+        partition_options: PartitionOptions,
+    ) -> (Self::ConstraintCommitment<E>, CompositionPoly<E>)
     where
-        E: FieldElement<BaseField = Self::BaseField>,
-    {
-        // first, build constraint composition polynomial from its trace as follows:
-        // - interpolate the trace into a polynomial in coefficient form
-        // - "break" the polynomial into a set of column polynomials each of degree equal to
-        //   trace_length - 1
-        let composition_poly = info_span!(
-            "build_composition_poly_columns",
-            num_columns = num_constraint_composition_columns
-        )
-        .in_scope(|| {
-            CompositionPoly::new(composition_poly_trace, domain, num_constraint_composition_columns)
-        });
-        assert_eq!(composition_poly.num_columns(), num_constraint_composition_columns);
-        assert_eq!(composition_poly.column_degree(), domain.trace_length() - 1);
-
-        // then, evaluate composition polynomial columns over the LDE domain
-        let domain_size = domain.lde_domain_size();
-        let composed_evaluations = info_span!("evaluate_composition_poly_columns").in_scope(|| {
-            RowMatrix::evaluate_polys_over::<DEFAULT_SEGMENT_WIDTH>(composition_poly.data(), domain)
-        });
-        assert_eq!(composed_evaluations.num_cols(), num_constraint_composition_columns);
-        assert_eq!(composed_evaluations.num_rows(), domain_size);
-
-        // finally, build constraint evaluation commitment
-        let constraint_commitment = info_span!(
-            "compute_constraint_evaluation_commitment",
-            log_domain_size = domain_size.ilog2()
-        )
-        .in_scope(|| {
-            let commitment = composed_evaluations
-                .commit_to_rows::<Self::HashFn, Self::VC>(self.options().partition_options());
-            ConstraintCommitment::new(composed_evaluations, commitment)
-        });
-
-        (constraint_commitment, composition_poly)
-    }
+        E: FieldElement<BaseField = Self::BaseField>;
```

Comment:
I would move this method into the "Required methods" section (i.e., after line 204).